src/lib/ndpi_main.c

```diff
@@ -6247,6 +6249,12 @@ u_int ndpi_get_ndpi_detection_module_size() {
   return(sizeof(struct ndpi_detection_module_struct));
 }
 
+void ndpi_set_debug_bitmask(struct ndpi_detection_module_struct *ndpi_str, NDPI_PROTOCOL_BITMASK debug_bitmask) {
+#ifdef NDPI_ENABLE_DEBUG_MESSAGES
```

Comment:
I would recommend to put the preprocessor conditional one line up as this function is only mandatory if `NDPI_ENABLE_DEBUG_MESSAGES` enabled.

Reply:
It is matter of taste. I think that having an ABI that doesn't depend too much on defines is preferable from an user point-of-view.
In the same .c file, set_ndpi_debug_function() has been already implemented with the same logic (I originally copied-and-pasted from it) but I see that the library is not always consistent about that.
No strong opinion